modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/PackageLoader.java

```diff
@@ -143,16 +147,31 @@ private BLangPackage loadPackage(PackageID pkgId, PackageEntity pkgEntity) {
             return null;
         }
 
+        String sourceName = null;
         if (pkgEntity.getKind() == PackageEntity.Kind.SOURCE) {
             pkgNode = this.sourceCompile((PackageSource) pkgEntity);
             pSymbol = symbolEnter.definePackage(pkgNode);
             pkgNode.symbol = pSymbol;
+            sourceName = ((PackageSource) pkgEntity).getPackageSourceEntries().get(0).getEntryName();
         } else {
             // This is a compiled package.
             // TODO Throw an error. Entry package cannot be a compiled package
             throw new RuntimeException("TODO Entry package cannot be a compiled package");
         }
 
+        if (!isValidPackage(pkgId, pkgNode)) {
```

Comment:
A block of code like this should be added to a separate private method, such as "validatePackage()", the main method needs to be kept at a higher level set of instructions, so it's easier to understand.